c/datasetjson.c

```diff
@@ -2336,30 +2449,47 @@ void streamDatasetForCopyAndRespond(HttpResponse *response, char *sourceDataset,
     zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_WARNING,  "ICSF error for SHA etag init for write, %d\n",rcEtag);
   }
 
-  int bufferSize = recordLength+1;
+  if(isTargetMember) {
+    int targetRecordLen = getTargetDsnRecordLength(targetDataset);
+    if(targetRecordLen < sourceRecordLen) {
+      fclose(inDataset);
+      fclose(outDataset);
+      respondWithError(response, HTTP_STATUS_INTERNAL_SERVER_ERROR, "Cannot copy dataset. Record length for target dataset is shorter than the source");
+      rc = deleteDatasetOrMember(response, targetDataset, responseMessage, &responseCode);
+      return ERROR_COPYING_DATASET;
+    }
+  }
+
+  int bufferSize = sourceRecordLen+1;
   char buffer[bufferSize];
   int bytesRead = 0;
   int bytesWritten = 0;
   int recordsWritten = 0;
 
   while (!feof(inDataset)){
-    bytesRead = fread(buffer,1,recordLength,inDataset);
+    bytesRead = fread(buffer,1,sourceRecordLen,inDataset);
```

Comment:
when source record length is less than target, on each line it would be best to right-pad with 0x40 (space) during the `fwrite`